packages/now-next/src/index.ts

```diff
@@ -231,24 +231,32 @@ export const build = async ({
     });
   }
 
-  const nowJsonPath = await findUp(['now.json', 'vercel.json'], {
-    cwd: path.join(workPath, path.dirname(entrypoint)),
-  });
-
   let hasLegacyRoutes = false;
+  let possibleNowJsonPath = entryPath;
   const hasFunctionsConfig = !!config.functions;
 
-  if (nowJsonPath) {
-    const nowJsonData = JSON.parse(await readFile(nowJsonPath, 'utf8'));
+  while (possibleNowJsonPath.length >= workPath.length) {
+    const curFiles = await readdir(possibleNowJsonPath);
+    const nowJsonFile = curFiles.find(
+      (file) => file.endsWith('now.json') || file.endsWith('vercel.json')
```

Comment:
Why the choice of endsWith here? Seems like it should be strict equality.

Reply:
Was left over from the `find-up` implementation, updated